packages/backend/src/peripherals/database/L2TransactionRepository.ts

```diff
@@ -44,7 +44,7 @@ export class L2TransactionRepository extends BaseRepository {
     const { starkKeyA, starkKeyB, data } = encodeL2TransactionData(record.data)
 
     const count = await this.countByTransactionId(record.transactionId)
```

Comment:
I'm worried that this is too hard on database, especially that we will be getting high load of transactions. Adding needs to be fast.

So either: pass altIndex to the function, or revisit idea of primary key.

Reply:
We can not pass altIndex to function in case of transaction API.

Reply:
in this case I'm leaning towards using primary key. Calling count() and then insert() in sequence is inefficient, and unsafe in case of asynchronous processes modifying this table (e.g. we don't have any guarantee that between calculating count and adding row, the count doesn't change). I know it's all edge cases and fast enough, but given two options:

1. using primary key = simple and easy code, no race conditions
2. using alt_id = more complex, inefficient code with race conditions

why would we choose 2? :)

Reply:
now I've realised that we can't escape looking into the table before adding a new record, because we need to mark the new record as a replacement and the previous one as replaced. So solution 1. above isn't simpler.